src/script/cpp_api/s_mainmenu.cpp

```diff
@@ -76,3 +76,25 @@ void ScriptApiMainMenu::handleMainMenuButtons(const StringMap &fields)
 	PCALL_RES(lua_pcall(L, 1, 0, error_handler));
 	lua_pop(L, 1); // Pop error handler
 }
+
+void ScriptApiMainMenu::beforeClose()
+{
+	SCRIPTAPI_PRECHECKHEADER
+
+	int error_handler = PUSH_ERROR_HANDLER(L);
+
+	// Get handler function
+	lua_getglobal(L, "core");
+	lua_getfield(L, -1, "on_before_close");
+	lua_remove(L, -2); // Remove core
+	if (lua_isnil(L, -1)) {
+		lua_pop(L, 1); // Pop callback
```

Comment:
`2`, for error handler?

Reply:
Right. The other functions were also wrong.
Fixed.

Reply:
If you wanted to clean up this even more, consider moving this code into a helper function such that all code is equally correct (or buggy).

Reply:
No thx. :)